core/server/master/src/main/java/alluxio/master/file/replication/ReplicationChecker.java

```diff
@@ -137,7 +137,8 @@ public ReplicationChecker(InodeTree inodeTree, BlockMaster blockMaster,
   @Override
   public void heartbeat() throws InterruptedException {
     // skips replication in safe mode when not all workers are registered
-    if (mSafeModeManager.isInSafeMode()) {
+    // skips replication check when there are no worker.
+    if (mSafeModeManager.isInSafeMode() || mBlockMaster.getWorkerCount() == 0) {
```

Comment:
Please add a debug log for better tracing, just in case
```
LOG.debug("Skip the ReplicationChecker in safe mode and when there are no workers");
```